awscli/customizations/s3/subcommands.py

```diff
@@ -818,9 +789,46 @@ class RbCommand(S3TransferCommand):
         "deleted."
     )
     USAGE = "<S3Uri>"
-    ARG_TABLE = [{'name': 'paths', 'nargs': 1, 'positional_arg': True,
+    ARG_TABLE = [{'name': 'path', 'positional_arg': True,
                   'synopsis': USAGE}, FORCE]
 
+    def _run_main(self, parsed_args, parsed_globals):
+        super(RbCommand, self)._run_main(parsed_args, parsed_globals)
+
+        if not parsed_args.path.startswith('s3://'):
+            raise TypeError("%s\nError: Invalid argument type" % self.USAGE)
+        bucket, key = split_s3_bucket_key(parsed_args.path)
+
+        if parsed_args.force:
+            # TODO: always validate or never validate
```

Comment:
Yeah this is interesting. So after thinking about it, I think we should apply the validation to both for consistency reasons. For the scenario where the person specified``s3://bucket/key`` for ``rb``, it is probably most likely they meant to specify ``rm``, in which case the ``rb`` command will fail. The only way this changes behavior is if the bucket was empty in the first place.